llvm/test/CodeGen/SPIRV/hlsl-intrinsics/log10.ll

```diff
@@ -0,0 +1,42 @@
+; RUN: llc -O0 -mtriple=spirv-unknown-linux %s -o - | FileCheck %s
+
+; CHECK: OpExtInstImport "GLSL.std.450"
+
+; CHECK: %[[#float:]] = OpTypeFloat 32
```

Comment:
Shall this test also validate other float sizes? Making sure we don't use the wrong type somewhere? (at least doubles?)

Reply:
The HLSL intrinsic is only valid on floats. In DXC bitcasting and warning about precision loss is done on the frontend to handle it if you do pass in a double, but it doesn't look like that's implemented in Clang yet. I did add a TODO since currently log10 (and many other math functions) pass legalization even though they shouldn't be generated by the frontend and they're not selectable on the backend.